core/src/main/java/org/elasticsearch/action/ActionModule.java

```diff
@@ -338,7 +344,18 @@ public ActionModule(boolean ingestEnabled, boolean transportClient, Settings set
         autoCreateIndex = transportClient ? null : new AutoCreateIndex(settings, clusterSettings, resolver);
         destructiveOperations = new DestructiveOperations(settings, clusterSettings);
         Set<String> headers = actionPlugins.stream().flatMap(p -> p.getRestHeaders().stream()).collect(Collectors.toSet());
-        restController = new RestController(settings, headers);
+        UnaryOperator<RestHandler> restWrapper = null;
+        for (ActionPlugin plugin : actionPlugins) {
+            UnaryOperator<RestHandler> newRestWrapper = plugin.getRestHandlerWrapper(threadPool.getThreadContext());
+            if (newRestWrapper != null) {
+                logger.debug("Using REST wrapper from plugin " + plugin.getClass().getName());
+                if (restWrapper != null) {
```

Comment:
> shall we check in the community which plugins may be using rest filters first?

Can you name a single plugin out there that implements a RestFilter? Let alone one that would work composed with another plugin implementing one? I searched for "implements RestFilter handleRequest" (the handleRequest is to distinguish from some other http libraries). All I see are forks of elasticsearch, no actual RestFilter implementations.

I'm in no rush to get this in, but I also don't think such a low level change to how plugins operate should have to wait for 6.0. If we did break such a use case, we could always implement support at that time if it was truly necessary and have in the next minor version. But I don't think we should overengineer something for what seems to me to not only be an esoteric and _theoretical_ case.